pkg/controller/garbagecollector/metrics/metrics.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const GarbageCollectorControllerSubsystem = "garbagecollector_controller"
+
+var (
+	GarbageCollectorResourcesSyncError = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Subsystem:      GarbageCollectorControllerSubsystem,
```

Comment:
it would be better to have `garbagecollector` as `Namespace` and `controller` as `Subsystem` in the options. The result will be the same, but it will be easier to differentiate subsystems from the same namespace.

Reply:
Your change seems a bit different from what I was expecting before. Also note that the name of the metric will end up being: Subsystem + Namespace + Name, so you need to remove the `garbagecollector` prefix from the metric Name.

Reply:
> Your change seems a bit different from what I was expecting before. 

I have reverted it back because of https://github.com/kubernetes/kubernetes/pull/106844#discussion_r782612711 

> Currently the metric will be named garbagecollector_controller_garbagecollector_resources_sync_error_total

Yup, that makes sense not to have it redundant.